tests/integration/test_wpk/test_wpk_manager.py

```
@@ -0,0 +1,844 @@
+    return json.loads(response.decode())
+
+
+def clean_logs():
+    truncate_file(LOG_FILE_PATH)
+
+
+def wait_download(line):
+    if 'Downloading WPK file from:' in line:
+        return line
+    return None
+
+def wait_downloaded(line):
+    if ('Download' in line) and ('finished' in line):
+        return line
+    return None
+
+def wait_chunk_size(line):
+    if ('Sending message to agent:' in line) and ('com write ' in line):
+        return line
+    return None
+
+def wait_wpk_custom(line):
+    if ('Sending message to agent:' in line) and ('com upgrade ' in line):
+        return line
+    return None
+
+
+def remove_wpk_package():
+    for file in os.scandir(UPGRADE_PATH):
+        if file.name.endswith('.wpk'):
+            os.unlink(file.path)
+
+def create_wpk_custom_file(file):
+    with open(file, 'wb') as f:
+        f.seek(1024*128)
+        f.write(b'\0')
+
+def get_sha_list(metadata):
+    global valid_sha1_list
+    agent_os = metadata['agents_os']
+    protocol = 'https://'
+    wpk_repo = metadata.get('wpk_repository')
+    architecture = 'x86_64'
+
+    if metadata.get('message_params') and metadata.get('message_params').get('version'):
+        agent_version = metadata.get('message_params').get('version')
+    else :
+        agent_version = MANAGER_VERSION
+    
+    if metadata.get('message_params') and metadata.get('message_params').get('use_http'):
+        protocol = 'http://' if metadata.get('message_params').get('use_http') == 1 else 'https://'
+    
+    # Generating file name
+    wpk_file = "wazuh_agent_{0}_linux_{1}.wpk".format(agent_version, architecture)
+    wpk_url = protocol + wpk_repo + "linux/" + architecture + "/" + wpk_file
+
+    wpk_file_path = os.path.join(UPGRADE_PATH, wpk_file)
+    
+    if not os.path.exists(wpk_file_path) and (not valid_sha1_list.get(wpk_file)):
+        try:
+            result = requests.get(wpk_url)
+        except requests.exceptions.RequestException as e:
+            pass
```

Maybe we should raise an exception reporting that the URL could not be obtained